streams/src/test/java/org/apache/kafka/streams/processor/internals/StateDirectoryTest.java

```diff
@@ -266,4 +268,51 @@ public void shouldUnlockGlobalStateDirectory() throws Exception {
         }
     }
 
+    @Test
+    public void shouldNotLockStateDirLockedByAnotherThread() throws IOException, InterruptedException {
```

Comment:
You don't count for this case -- you wrote it. ;P \cc @enothereska @bbejeck @guozhangwang